client/state/data-layer/wpcom/activity-log/rewind/to/index.js

```diff
@@ -10,24 +11,47 @@ import {
 	REWIND_RESTORE,
 } from 'state/action-types';
 import {
-	rewindCompleteRestore,
 	rewindRestoreUpdateError,
+	getRewindRestoreProgress,
 } from 'state/activity-log/actions';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { http } from 'state/data-layer/wpcom-http/actions';
+
+const debug = debugFactory( 'calypso:data-layer:activity-log:rewind:to' );
+
+const fromApi = data => ( {
+	restoreId: +data.restore_id,
+} );
+
 const requestRestore = ( { dispatch }, action ) => {
 	dispatch( http( {
+		apiVersion: '1',
 		method: 'POST',
 		path: `/activity-log/${ action.siteId }/rewind/to/${ action.timestamp }`,
-		apiVersion: '1',
 	}, action ) );
 };
 
-export const receiveRestoreSuccess = ( { dispatch }, { siteId, timestamp } ) => {
-	dispatch( rewindCompleteRestore( siteId, timestamp ) );
+export const receiveRestoreSuccess = ( { dispatch }, { siteId, timestamp }, next, apiData ) => {
+	const { restoreId } = fromApi( apiData );
+	if ( restoreId ) {
+		debug( 'Request restore success, restore id:', restoreId );
+		dispatch( getRewindRestoreProgress( siteId, timestamp, restoreId ) );
+	} else {
+		debug( 'Request restore response missing restore_id' );
+		dispatch( rewindRestoreUpdateError(
+			siteId,
+			timestamp,
+			{
+				// FIXME: should include status?
```

Comment:
Is this missing `restoreId` case something we have reason to expect? If not, I'm not too fussed what we do here. Seems only one of many things that could go wrong. But, yeah, `finished` would make sense, since nothing else will be happening.